### PR TITLE
Display citizen initiatives in the event list

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/SearchController.php [hunk 1]
```diff
@@ -30,6 +30,7 @@ public function searchEventsAction(Request $request)
         try {
             $results = $this->get('app.search.search_results_provider')->find($search);
         } catch (GeocodingException $exception) {
+            dump($exception->getMessage());die;
```
Comment (hhamon) [inline src/Controller/EnMarche/SearchController.php]: debug statement to be removed!

File: src/DataFixtures/ORM/LoadCitizenInitiativeData.php [hunk 2]
```diff
@@ -106,6 +108,22 @@ public function load(ObjectManager $manager)
...
+            'description' => 'Nous allons rendre Kilchberg propre',
+            'address' => PostAddress::createForeignAddress('CH', '8802', 'Kilchberg', '54 Pilgerweg', 47.3164934, 8.553012),
+            'begin_at' => new \DateTime(date('Y-m-d', strtotime('+11 days')).' 09:00:00'),
+            'finish_at' => new \DateTime(date('Y-m-d', strtotime('+11 days')).' 12:00:00'),
+            'expert_assistance_needed' => false,
+            'expert_assistance_description' => '',
+            'coaching_requested' => false,
+        ]);
+        $initiative5->incrementParticipantsCount();
+        $initiative5->setInterests(['environement']);
```
Comment (hhamon) [inline src/DataFixtures/ORM/LoadCitizenInitiativeData.php]: `/environement/environment/` ou `environnement`  en français

File: src/Doctrine/Hydrators/EventHydrator.php [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        foreach ($this->_stmt->fetchAll(\PDO::FETCH_ASSOC) as $row) {
+            $this->hydrateRowData($row, $result);
+        }
+
+        return $result;
+    }
+
+    protected function hydrateRowData(array $row, array &$result)
+    {
+        if (count($row) == 0) {
```
Comment (hhamon) [inline src/Doctrine/Hydrators/EventHydrator.php]: `if (!count($row)) {`

File: src/Doctrine/Hydrators/EventHydrator.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            return false;
+        }
+
+        $uuid = Uuid::fromString($row['uuid']);
+        $organizer = $this->_em->getRepository(Adherent::class)->findOneBy(['id' => $row['organizer_id']]);
```
Comment (hhamon) [inline src/Doctrine/Hydrators/EventHydrator.php]: You'll make a lot of nested queries with this.

File: src/Doctrine/Hydrators/EventHydrator.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        $committee = $this->_em->getRepository(Committee::class)->findOneBy(['id' => $row['committee_id']]);
```
Comment (hhamon) [inline src/Doctrine/Hydrators/EventHydrator.php]: Same here.

File: src/Doctrine/Hydrators/EventHydrator.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        } else {
+            $address = PostAddress::createForeignAddress($row['address_country'], $row['address_postal_code'], $row['address_city_name'], $row['address_address'], $row['address_latitude'], $row['address_longitude']);
+        }
+
+        $event = new Event(
+            $uuid,
+            $organizer,
+            $committee,
+            $row['name'],
+            new EventCategory(),
```
Comment (hhamon) [inline src/Doctrine/Hydrators/EventHydrator.php]: Empty category?

File: src/Doctrine/Hydrators/EventHydrator.php (hunk 3 above)
Comment (HeahDude) [inline src/Doctrine/Hydrators/EventHydrator.php]: I would say `if (!$row)` or keep this check but make it strict.

File: src/Controller/EnMarche/CitizenInitiativeController.php [hunk 7]
```diff
@@ -20,7 +22,7 @@
  */
 class CitizenInitiativeController extends Controller
 {
-    use EntityControllerTrait;
+    use EntityControllerTrait, CanaryControllerTrait;
```
Comment (HeahDude) [inline src/Controller/EnMarche/CitizenInitiativeController.php]: Please add a new line.

**Pull request merged**